framework/web_app/src/pages/StockOverview.vue

```diff
@@ -1,96 +1,260 @@
+                        clickable
+                        v-close-popup
+                        v-for="level in stockLevelStore.stockLevels"
+                    >
+                        <q-item-section avatar>
+                            <q-avatar
+                                :color="
+                                    getStockLevelColour(level.stock_level_id)
+                                "
+                                size="25px"
+                            />
+                        </q-item-section>
+
+                        <q-item-section>
+                            <q-item-label>{{ level.description }}</q-item-label>
+                        </q-item-section>
+                    </q-item>
+                </q-btn-dropdown>
+                <q-btn
+                    class="q-mx-sm"
+                    @click="shouldDisplayAddToShoppingCartModal = true"
+                    flat
+                    icon="shopping_cart"
+                    rounded
+                />
+
+                <q-separator
+                    class="q-ml-sm q-mr-md"
+                    vertical
+                />
+                <p class="text-bold q-ma-sm">{{ item.name }}</p>
+            </q-card-actions>
+
+            <q-card-section class="col">
+                <p class="text-weight-bold q-ma-none">Location</p>
+                <p class="q-ma-none">{{ 'item.stock_location' }}</p>
+            </q-card-section>
+
+            <q-card-section class="col">
+                <p class="text-weight-bold q-ma-none">Stock Group</p>
+                <p class="q-ma-none">{{ 'item.stock_group' }}</p>
+            </q-card-section>
+        </q-card-section>
+    </q-card>
+
+    <q-dialog v-model="shouldDisplayCreateStockItemModal">
+        <q-card
+            class="q-pa-md"
+            style="width: 700px; max-width: 80vw"
+        >
+            <p class="text-h4">Add a new stock item</p>
+            <q-form
+                class="q-gutter-md"
+                @submit="
+                    createStockItemAsync(createStockItemForm).then(
+                        clearCreateStockItemForm
+                    )
+                "
+            >
+                <q-input
+                    :rules="[
+                        (val: string) =>
+                            (val && val.length > 0) || 'Please type something'
+                    ]"
+                    autofocus
+                    filled
+                    label="Name"
+                    v-model="createStockItemForm.name"
+                />
+
+                <q-input
+                    type="number"
+                    filled
+                    hint="Alerts disabled if set to zero"
+                    label="Days Until Stocktake Alert"
+                    v-model="createStockItemForm.days_until_stocktake_alert"
+                />
+
+                <q-select
+                    :option-label="(opt: StockLevel) => opt.description"
```

Comment:
**Move all inline functions into the script.**

**Quasar:**
![image](https://github.com/BenTalese/DiscountDora/assets/95552720/b86eecc0-0f89-4819-a7cd-0fa8427d027d)

**GPT explanation:** 

Garbage Collection:

Inline function definitions generate additional objects that need to be garbage collected, increasing the load on the garbage collector and potentially leading to performance issues.
Using referenced functions minimizes the number of objects created, leading to less frequent garbage collection and better performance.
Reactivity System in Vue:

Vue's reactivity system tracks changes in the data and updates the DOM accordingly.
Inline functions create new function references on each render, which can cause Vue to re-evaluate and update the DOM more than necessary.
Referenced functions maintain consistent references, helping Vue to optimize updates and avoid unnecessary re-renders.
Event Handling:

Inline functions lead to the creation of new event handlers on each render, which can be inefficient.
By using referenced functions, the same event handler is reused, resulting in more efficient event handling.